examples/datasource-http-backend/pkg/plugin/datasource.go

```diff
@@ -0,0 +1,152 @@
+// since otherwise we will only get a not implemented error response from plugin in
+// runtime. In this example datasource instance implements backend.QueryDataHandler,
+// backend.CheckHealthHandler interfaces. Plugin should not implement all these
+// interfaces- only those which are required for a particular task.
+var (
+	_ backend.QueryDataHandler      = (*Datasource)(nil)
+	_ backend.CheckHealthHandler    = (*Datasource)(nil)
+	_ instancemgmt.InstanceDisposer = (*Datasource)(nil)
+)
+
+// NewDatasource creates a new datasource instance.
+func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	opts, err := settings.HTTPClientOptions()
+	if err != nil {
+		return nil, fmt.Errorf("http client options: %w", err)
+	}
+	cl, err := httpclient.New(opts)
+	if err != nil {
+		return nil, fmt.Errorf("httpclient new: %w", err)
+	}
+	return &Datasource{
+		settings:   settings,
+		httpClient: cl,
+	}, nil
+}
+
+// Datasource is an example datasource which can respond to data queries, reports
+// its health and has streaming skills.
+type Datasource struct {
+	settings backend.DataSourceInstanceSettings
+
+	httpClient *http.Client
+}
+
+// Dispose here tells plugin SDK that plugin wants to clean up resources when a new instance
+// created. As soon as datasource settings change detected by SDK old datasource instance will
+// be disposed and a new one will be created using NewSampleDatasource factory function.
+func (d *Datasource) Dispose() {
+	// Clean up datasource instance resources.
+	d.httpClient.CloseIdleConnections()
+}
+
+// QueryData handles multiple queries and returns multiple responses.
+// req contains the queries []DataQuery (where each query contains RefID as a unique identifier).
+// The QueryDataResponse contains a map of RefID to the response for each query, and each response
+// contains Frames ([]*Frame).
+func (d *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// create response struct
+	response := backend.NewQueryDataResponse()
+
+	// loop over queries and execute them individually.
+	for _, q := range req.Queries {
+		res, err := d.query(ctx, req.PluginContext, q)
+		if err != nil {
+			return nil, fmt.Errorf("query: %w", err)
+		}
+		// save the response in a hashmap
+		// based on with RefID as identifier
+		response.Responses[q.RefID] = res
+	}
+
+	return response, nil
+}
+
+func (d *Datasource) query(ctx context.Context, pCtx backend.PluginContext, query backend.DataQuery) (backend.DataResponse, error) {
+	// Response to be returned.
+	var response backend.DataResponse
+
+	// Do HTTP request
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, d.settings.URL, nil)
+	if err != nil {
+		return response, fmt.Errorf("new request with context: %w", err)
+	}
+	resp, err := d.httpClient.Do(req)
+	if err != nil {
+		return response, fmt.Errorf("http client do: %w", err)
+	}
+	defer resp.Body.Close()
```

Comment:
Great. But would also like you to handle the case if `Body.Close` returns an error and log that. Otherwise those errors just being swallowed/ignored which is not great.